ios/browser/brave_ads/ads_service_impl_ios.mm

```diff
@@ -513,4 +534,26 @@
   InitializeAds(std::move(callback));
 }
 
+void AdsServiceImplIOS::PrefetchNewTabPageAdCallback(
+    base::optional_ref<const NewTabPageAdInfo> new_tab_page_ad) {
+  CHECK(!prefetched_new_tab_page_ad_);
+  CHECK(is_prefetching_new_tab_page_ad_);
+
+  is_prefetching_new_tab_page_ad_ = false;
+
+  if (new_tab_page_ad) {
+    prefetched_new_tab_page_ad_ = *new_tab_page_ad;
+  }
+}
+
+void AdsServiceImplIOS::OnParseAndSaveCreativeNewTabPageAdsCallback(
+    ParseAndSaveCreativeNewTabPageAdsCallback callback,
+    bool success) {
+  if (success) {
+    PrefetchNewTabPageAd();
```

Comment:
See `AdsServiceImpl::OnResourceComponentDidChange` where we also `PrefetchNewTabPageAd()` after the resource is updated?

Reply:
Ok this makes sense.